contracts/NFTVault.sol

```diff
@@ -103,7 +109,7 @@ contract NFTVault is IERC721Receiver, IERC1155Receiver, AccessControl {
     {
         require(keys.balanceOf(msg.sender) != 0, "sender cannot have 0 keys");
         require(nfts.length > 0, "there are no NFTs in the contract");
-        require(amount >= 10, "amount must be atleast 10");
+        require((amount/BASE_UNIT) >= 10, "amount must be atleast 10");
```

Comment:
Probably better than having division there, it would be better to compare `amount` to `10*BASE_UNIT`